index.js

```diff
@@ -1,13 +1,39 @@
 const url = require('url');
 const Promise = require('bluebird');
 const net = require('net');
+const os = require('os');
 
 Promise.config({
   cancellation: true
 });
 
-async function checkInternetConnected (config = {}) {
-  const {timeout = 5000, retries = 5, domain = 'https://apple.com'} = config;
+/**
+ * @Laurix1983 added
+ * Helper function to try to search interface address
+ */
+function getInterfaceAddress(intr, family) {
+
+
+  const allInterfaces = os.networkInterfaces();
+
+  const foundIf = allInterfaces[intr];
+  if (foundIf) {
+    if (!family)
+      family = 'IPv4'
+    const foundFamily = foundIf.find(f => f.family == family)
+    if (foundFamily.address)
+      return foundFamily.address;
+
+
+
+  }
+
+  return null;
+
+}
+
+async function checkInternetConnected(config = {}) {
+  const { timeout = 5000, retries = 5, domain = 'https://apple.com', network_interface = undefined, family = undefined } = config;
```

Comment:
Codacy has a fix for the issue: [Replace `·timeout·=·5000,·retries·=·5,·domain·=·'https://apple.com',·network_interface·=·undefined,·family·=·undefined` with `⏎····timeout·=·5000,⏎····retries·=·5,⏎····domain·=·"https://apple.com",⏎····network_interface·=·undefined,⏎····family·=·undefined,⏎·`](https://app.codacy.com/gh/aankur/check-internet-connected/pullRequest?prid=9568874)
```suggestion
  const {
```